### ISSUE TITLE
srv — importable is a find_spec result for kokoro/qwen/whisper too, and the name overclaims

### ISSUE THREAD

**Issue by dudarenok-maker** [dudarenok-maker/Castwright#1965]
## What

#1963 fixed Coqui's `livePackageImportable` to prefer the sticky `coqui_import_ok` over the `find_spec` probe. That closes the honesty gap for **one** engine. The investigation done alongside it found the same shape everywhere else, plus a naming problem the fix does not remove.

**1. Kokoro, Qwen and Whisper have no real-import counterpart at all.**

Sidecar side — all three are `importlib.util.find_spec(...)` only, with no `*_import_ok` sticky sibling:

- `server/tts-sidecar/main.py` — `_qwen_package_installed`, `_kokoro_package_installed`, `_whisper_package_installed`

Node side — the same "advertises importable from a probe that never imports" shape #1944/#1963 fixed for Coqui:

- `server/src/tts/voice-engine-registry.ts` — kokoro's `livePackageImportable: (h) => h.kokoroPackageInstalled`, qwen's `(h) => h.qwenPackageInstalled`

Coqui is the engine where this bit, because the speechbrain lazy-proxy collision (#1944) made a `find_spec`-installed package genuinely unimportable. Nothing makes the other three immune — they simply have not hit an equivalent yet. Whether they *should* get a sticky import signal is a real decision: it costs a real import attempt to populate, which is exactly why Coqui's is sticky-null until something tries.

**2. `importable` still overclaims, even after #1963.**

`EngineProbeResult.importable` (`server/src/tts/models-status.ts`) is named for a guarantee it only sometimes provides:

- for kokoro/qwen it is *always* a bare `find_spec` result;
- for coqui it is `find_spec` whenever `coqui_import_ok` is still `null` — which is the common pre-import-attempt case, and the most common wire value.

So a client reading `importable: true` cannot tell whether it means "we imported it" or "a spec exists for it". #1963 deliberately left this alone: renaming a client-facing field is a contract decision, not a drive-by.

## Why it matters

#1944's on-box run is the worked example: `/health` said `coqui_package_installed: true`, a real `from TTS.api import TTS` raised, and that false signal caused a whole section of the fs-38 acceptance run to be scoped as unblocked when it was not. Any consumer trusting `importable` for the three unfixed engines can still repeat that mistake, and any consumer trusting it for Coqui *before* a real import attempt can too.

## Suggested shape

Two separable pieces — the second is worth doing even if the first is declined:

1. Decide whether kokoro/qwen/whisper get sticky `*_import_ok` fields, weighing the cost of the import attempt that populates them. A cheaper alternative worth considering: populate the sticky field opportunistically on the first real load rather than probing eagerly, which is what Coqui already does.
2. Make the wire field say what it means. Either rename `importable` to something that describes a spec probe, or keep the name and make it genuinely mean "a real import succeeded" — with a separate field for the spec result. Either way `openapi.yaml` and the frontend consumers move with it.

## Notes

Found by the out-of-scope investigation requested during #1963, which is the follow-up to #1944 / PR #1962. Deliberately not fixed there — #1963's whole point was to leave the `null` path byte-for-byte unchanged, and both of these change behaviour or contract.

**Comment by dudarenok-maker** [comment]
## One more surface for the same naming problem — `models-inventory`'s `packageInstalled`

The independent review on PR #1970 (finding L5) turned up a third instance, worth folding into this ticket since it is the same decision.

After #1963, `server/src/routes/models-inventory.ts` computes coqui's `packageInstalled` as `sidecar.coquiImportOk ?? pkgInstalled(...)`. That is the correct honesty fix, but it means an **installed-but-unimportable** Coqui — precisely the #1944 speechbrain-proxy state — now surfaces as `installState: 'package-missing'`.

The field name says "installed on disk". The value now says "importable". Those come apart in exactly the case the field exists to describe, and the user-facing consequence points the wrong way: someone seeing "package missing" reinstalls, when the actual remedy is evicting the speechbrain lazy proxies.

Three reasons it was left as-is rather than blocking that PR, all of which still argue for deciding it here rather than reverting it:

- `deriveCoquiInstallState` (`server/src/routes/sidecar-health.ts`) already set this precedent in #1962 — `coqui_import_ok === false` maps to `not-installed` — so #1963 made two surfaces consistent rather than inventing a new inconsistency.
- There is currently **no frontend consumer** of `packageInstalled` or `importable` anywhere in `src/`, so nothing renders the misleading label today.
- Reverting it would restore the honesty gap #1963 exists to close.

So this is not a bug to undo; it is the same contract question this issue already asks, on a third field. Whatever naming this ticket settles on should cover `packageInstalled` and `installState`'s `package-missing` value alongside `importable` — a state meaning "present but unusable" probably wants to be sayable, rather than collapsed into either "missing" or "ready".

**Comment by dudarenok-maker** [comment]
## Implementation brief

Decisions are settled. No plan doc under `docs/features/` — this is a localized correctness fix across two existing surfaces, not a feature; this comment plus the paired tests are the spec.

### One premise in the issue body is wrong — read this first

**`importable` never crosses the wire.** Verified: zero occurrences in `openapi.yaml`, zero in `src/`. The client-facing field is `EngineStatus.packageBroken` (`openapi.yaml` ~7312), and `server/src/tts/models-status.ts:64` is `importable`'s only reader. So "renaming a client-facing field is a contract decision" — the reasoning that kept this out of #1963 — does not apply. It is an internal server type field: 3 production files, 3 test files, no `openapi:types` regen, no frontend change.

Everything below is scoped on that basis.

### Decision 1 — split `importable` into two fields

Not a rename. `EngineProbeResult` gets:

```ts
importOk: boolean | undefined;    // a REAL import attempt succeeded/failed
specPresent: boolean | undefined; // the find_spec probe
```

and `packageBroken` derives from both:

```ts
const packageBroken =
  p.packageOnDisk && (p.importOk === false || (p.importOk === undefined && p.specPresent === false));
```

**This predicate is exactly equivalent to today's** `packageOnDisk && (importOk ?? specPresent) === false`. Landing the split on its own must be a pure refactor with **no behaviour change** — that is the property to test first, before any sidecar work shifts what the inputs contain.

`undefined` keeps its documented meaning: unknown (sidecar down / older sidecar), never "broken". See the fail-open rule at `voice-engine-registry.ts:26-29` and the "Honesty limit" section of `docs/superpowers/specs/2026-07-14-wizard-models-status-and-recommendation-design.md`.

### Decision 2 — sticky `*_import_ok` for kokoro, qwen AND whisper

Each has exactly one import chokepoint, already inside a `try/except`. Opportunistic only — record the outcome of an import that was happening anyway. **No eager probe.** Coqui's startup pin costs a measured +11.7 s of sidecar-unreachable boot; nobody wants that per engine.

| Engine | Function | Import |
|---|---|---|
| kokoro | `KokoroEngine._ensure_loaded` (~2695) | `from kokoro_onnx import Kokoro` (~2707) |
| qwen | `QwenEngine._load_qwen_model` (~4326) | `from qwen_tts import Qwen3TTSModel` (~4358) |
| whisper | `WhisperEngine._ensure_loaded` (~6206) | `from faster_whisper import WhisperModel` (~6217) |

Copy the `_COQUI_IMPORT_OK` template (`main.py:1230`, setter `:1233`, four write sites at `:1523`, `:1535`, `:7691`, `:7694`). Two properties of that template are load-bearing and **must** be copied:

1. **Catch `BaseException`, not `ImportError`.** All three sites today catch `ImportError` only. The second documented shape of the #1944 collision is a duplicate-kernel-registration `RuntimeError`; an `ImportError`-only record leaves the flag at `None` (or stale `True`) while the engine is genuinely unstartable. That is #1962's review finding 3, already fixed once for Coqui. Record, then **re-raise unchanged**.
2. **Set on success AND failure**, through a setter, never reset.

**Qwen needs care:** `import torch` and `from qwen_tts import ...` currently share one `try`. A torch failure would record `qwen_import_ok=False` and mis-attribute the fault. Separate them so the flag reflects `qwen_tts` specifically. Also note `_load_qwen_model` continues past the import into `from_pretrained` and a `.to(device)` retry loop with documented intermittent meta-tensor faults — so `qwen_import_ok` means strictly "the import statement returned", not "the engine loaded". Do not let a comment or field name overclaim a second time; that is the whole subject of this issue.

`_load_qwen_model` is the single chokepoint for all three Qwen models (`_ensure_base_loaded` at ~4539 and ~4564, `_ensure_design_loaded` at ~4927).

**Honest ceiling, worth a comment in the code:** a sticky flag only populates once something tries to load. On an install that never renders on Kokoro, `kokoro_import_ok` stays null and the find_spec fallback governs — the same residual window Coqui has, minus the startup pin. This narrows the hole; it does not close it.

### Node side

- `/health` gains `kokoro_import_ok`, `qwen_import_ok`, `whisper_import_ok` (alongside the existing `coqui_import_ok` at ~8339).
- `SidecarHealthResult` gains the three matching fields. **Normalise absent → `null` at that boundary**, verbatim as `sidecar-health.ts:421-424` already does for coqui, so every consumer sees one "no real import attempt to trust" value.
- `voice-engine-registry.ts` — replace the single `livePackageImportable` with the two accessors the split needs. Kokoro (`:53`) and qwen (`:64`) are currently bare `find_spec`; coqui (`:81`) is `h.coquiImportOk ?? h.coquiPackageInstalled`. After this change all engines have the same shape.
- `models-inventory.ts` — coqui at `:263` already does `sidecar.coquiImportOk ?? pkgInstalled(...)`; kokoro (`:151`), qwen (`:177`, `:203`, `:229`) and whisper (`:290`) do not. Make all six uniform through one helper rather than repeating the `??`.

### Decision 3 — `diagnostics.ts`, which no registry fix reaches

`server/src/routes/diagnostics.ts:153-157` reads `sidecar.kokoroPackageInstalled` / `sidecar.qwenPackageInstalled` **directly**, bypassing the registry accessor, then renders `reachable · <names> package not importable — repair in Model Manager` to the operator. Fixing the registry does not reach it.

- Route it through the same accessor so it inherits the real-import signal.
- **Keep Coqui out of that list**, and add a comment saying why: Coqui is opt-in (`requirements/nvidia-cuda.txt:4`, `amd-rocm.txt:7` — "install it from the Model Manager"), so including it would fire "Coqui package not importable" on every install that deliberately never installed it.
- The string overclaims in the other direction too: the filter is `e.pkg === false`, which means the package is **missing**, and the state passed is `'package-missing'` — but the copy says "not importable". Fix the wording to match what is actually detected.

### Two defects found in passing — fold both in, they are in code this change already touches

1. **`main.py:2422`, `CoquiEngine.clone_voice`** — does `import TTS as _tts_pkg` *before* `self._ensure_loaded(...)` on the next line, with no `_record_coqui_import_result`. A clone attempt against a broken Coqui therefore raises without ever recording the failure. That is exactly the #1944 shape, on a path that does not record it.
2. **The import-pin docstring, `main.py` ~7658** — justifies gating on weights rather than package with "`coqui-tts` ships as an ordinary dependency and is always present". It does not: `coqui-tts` appears in no requirements file's install lines, and both `nvidia-cuda.txt:4` and `amd-rocm.txt:7` say Coqui is opt-in. The gate still works by correlation, but the stated reason is wrong and would mislead anyone reasoning about the null window.

### Tests

- **pytest** (`server/tts-sidecar/tests/`) — each engine records `True` on a successful import and `False` on a failing one, the failure is **re-raised**, a non-`ImportError` failure is also recorded (the #1962 finding-3 shape), and `/health` carries all four keys. Follow `test_coqui_import_pin.py` / `test_runtime_wiring.py`.
- **vitest** — `sidecar-health.test.ts` (absent → `null` normalisation for all three), `voice-engine-registry.test.ts`, `models-status.test.ts` (**start here: the split is behaviour-preserving** — the existing 6 cases must pass unchanged), `models-inventory.test.ts`, `diagnostics.test.ts`.
- The regression test that matters most: **spec-present + import failed → `packageBroken: true`** for kokoro and qwen. That is the #1944 false signal, and it is what the Repair CTA in `src/components/{kokoro,qwen}-install.tsx:~105` hangs off.

### Also required

- **On-box acceptance row.** A package that is present but genuinely unimportable cannot be manufactured in CI. Add a row to `docs/testing/onbox-acceptance-register.md` per CLAUDE.md's before-shipping step 3 — and update the live HTML register via the URL in that file's header, never by publishing a fresh one.
- **Release notes, both files.** `docs/release-notes-next.md` (technical, PR-ref'd) and a brand-voice line in `RELEASE_NOTES.md` — this one *is* user-visible: Model Manager will now correctly offer Repair for a Kokoro or Qwen package that installs but will not import, where today it stays silent.
- `docs/wiki/Advanced-Settings.md` needs no row — no new registry knob.
